lib/_stream_transform.js

```diff
@@ -115,9 +103,16 @@ function Transform(options) {
 
   Duplex.call(this, options);
 
-  this._transformState = new TransformState(this);
-
-  var stream = this;
+  this._transformState = {
+    afterTransform: (er, data) => {
+      return afterTransform(this, er, data);
+    },
```

Comment:
As discussed and checked with @mcollina, this can be really fast using `bind`, especially once [this CL](https://codereview.chromium.org/2916063002/) lands in V8. The CL should apply cleanly to 6.0, but also 5.9.

Here's a simple micro-benchmark:

```javascript
function bar() {}

function A() { this.x = (x) => bar(x); }

function Ab() { this.x = bar.bind(undefined); }

function c() { return new A(); }

function cb() { return new Ab(); }

function l() { return {x: (x) => bar(x)}; }

function lb() { return {x: bar.bind(undefined)}; }

const N = 10000000;

function test(fn, n) {
	for (var i = 0; i < n; ++i) {
		fn.call(this);
	}
}

const FNS = [c, cb, l, lb];

for (const fn of FNS) {
	test(fn, 100);
}

for (const fn of FNS) {
	console.time(fn.name);
	test(fn, N);
	console.timeEnd(fn.name);
}
 ```